byterun/caml/osdeps.h

```diff
@@ -91,12 +91,29 @@ extern char_os * caml_executable_name(void);
 */
 extern char_os *caml_secure_getenv(char_os const *var);
 
-/* Windows Unicode support */
+#ifdef _WIN32
+extern void caml_probe_os_version(void);
+#else
+#define caml_probe_os_version()
+#endif
+
+#if WINDOWS_UNICODE && defined(_WIN32)
+extern void caml_setup_terminal(void);
+#else
+#define caml_setup_terminal()
+#endif
```

Comment:
I have not seen this kind of extern-or-define-away trick before, and I wonder whether it was completely safe -- or if there is a risk of, say, link-time incompatibilities caused by a runtime compiled with the version and an application compiled without.

Reply:
This is an internal function only, so I think it's fine - you'd never be linking with the runtime in order to get this function.

Reply:
This is minor, but why not rename `caml_probe_os_version` and `caml_setup_terminal` in `win32.c` to `caml_probe_win32_version` and `caml_setup_win32_terminal`, declare those functions in `osdeps.h` (at the end of the file there is a part protected by `#ifdef _WIN32`), and move the `#if`'s above to `caml_sys_init` below (without the `#else` part) ? Personally I find the flow easier to understand like that: it is clear that the code is only relevant for Win32 when looking at `caml_sys_init` and there is no need to use the `extern-or-define-away` idiom.

Reply:
@nojb - I'd be happy with that. @gasche - this came about through trying to step back from too much Windows-specific stuff in sys.c. Did I go too far? 🙂

Reply:
I have no strong opinion, and given that you and @nojb are likely to be in charge of understanding this code for the years to come it makes sense to pick something that you are both comfortable with.

Reply:
I am not a big fan of the `#define`-function-name-to-nothing trick, and I agree with @nojb's suggestion (assuming I understand it correctly).  Feel free to guard both declaration in osdeps.h and use in sys.c with `#ifdef _WIN32`.  Possibly, `caml_win32_setup_terminal` should be defined in win32.c regardless of `WINDOWS_UNICODE` (and do nothing if it's not defined).